File: backend/app/schemas/generation.py
from pydantic import BaseModel
from datetime import datetime


class GenerateRequest(BaseModel):
    extra_prompt: str | None = None


class TaskUpdateRequest(BaseModel):
    extra_prompt: str | None = None


class RegenerateRequest(BaseModel):
    extra_prompt: str | None = None


class SectionUpdate(BaseModel):
    content: str


class SectionResponse(BaseModel):
    id: int
    task_id: int
    section_key: str
    title: str
    content: str | None = None
    section_order: int
    status: str
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SectionProgressResponse(BaseModel):
    """轻量级章节进度响应（用于SSE推送，不包含完整内容）"""
    id: int
    section_key: str
    title: str
    section_order: int
    status: str

    model_config = {"from_attributes": True}


class TaskResponse(BaseModel):
    id: int
    application_id: int
    status: str
    extra_prompt: str | None = None
    generate_source_code: bool = True
    generate_db_design: bool = True
    generate_diagrams: bool = False
    total_sections: int
    completed_sections: int
    created_at: datetime
    updated_at: datetime
    sections: list[SectionResponse] = []

    model_config = {"from_attributes": True}


class TaskProgressResponse(BaseModel):
    """轻量级任务进度响应（用于SSE推送）"""
    id: int
    application_id: int
    status: str
    total_sections: int
    completed_sections: int
    updated_at: datetime
    sections: list[SectionProgressResponse] = []

    model_config = {"from_attributes": True}

File: frontend/src/api/generation.ts
import client from './client'

export interface GenerateRequest {
  extra_prompt?: string
}

export interface TaskUpdateRequest {
  extra_prompt?: string
}

export interface RegenerateRequest {
  extra_prompt?: string
}

export interface SectionUpdateRequest {
  content: string
}

export interface SectionRecord {
  id: number
  task_id: number
  section_key: string
  title: string
  content: string | null
  section_order: number
  status: string
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface TaskRecord {
  id: number
  application_id: number
  status: string
  extra_prompt: string | null
  generate_source_code: boolean
  generate_db_design: boolean
  total_sections: number
  completed_sections: number
  created_at: string
  updated_at: string
  sections: SectionRecord[]
}

export function startGeneration(appId: number, data: GenerateRequest = {}) {
  return client.post<TaskRecord>(`/applications/${appId}/generate`, data)
}

export function getGeneration(appId: number) {
  return client.get<TaskRecord>(`/applications/${appId}/generation`)
}

export function updateTask(appId: number, data: TaskUpdateRequest) {
  return client.put<TaskRecord>(`/applications/${appId}/generation`, data)
}

export function updateSection(sectionId: number, data: SectionUpdateRequest) {
  return client.put<SectionRecord>(`/generation-sections/${sectionId}`, data)
}

export function regenerateSection(sectionId: number, data: RegenerateRequest = {}) {
  return client.post<SectionRecord>(`/generation-sections/${sectionId}/regenerate`, data)
}

export function generateDiagrams(appId: number) {
  return client.post<TaskRecord>(`/applications/${appId}/generate-diagrams`)
}

const EXPORT_TIMEOUT = 300_000 // 5 minutes for document generation

export function exportManualWord(appId: number) {
  return client.get(`/applications/${appId}/export/manual/word`, { responseType: 'blob', timeout: EXPORT_TIMEOUT })
}

export function exportManualPdf(appId: number) {
  return client.get(`/applications/${appId}/export/manual/pdf`, { responseType: 'blob', timeout: EXPORT_TIMEOUT })
}

export function exportSourceCodeWord(appId: number) {
  return client.get(`/applications/${appId}/export/source-code/word`, { responseType: 'blob', timeout: EXPORT_TIMEOUT })
}

export function exportSourceCodePdf(appId: number) {
  return client.get(`/applications/${appId}/export/source-code/pdf`, { responseType: 'blob', timeout: EXPORT_TIMEOUT })
}

export function exportWord(appId: number) {
  return client.get(`/applications/${appId}/export/word`, { responseType: 'blob', timeout: EXPORT_TIMEOUT })
}

export function exportPdf(appId: number) {
  return client.get(`/applications/${appId}/export/pdf`, { responseType: 'blob', timeout: EXPORT_TIMEOUT })
}
